### PR TITLE
feat(player): implement Thumbnail component for player

### PR DESCRIPTION
**Pull request by Slashgear**
## Why

#1389 

## How

- add Thumbnail component
- add page of documentation in API, is this ok ? 

## Todo

- [x] documentation update
- [x] example in player-example package

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#1389: `<Thumbnail>` component](https://issuehunt.io/repos/274495425/issues/1389)
---
</details>

### CODE REVIEW

**Comment by Slashgear** [comment]
No worry @JonnyBurger ! 

Answering those question would help me a lot on figuring if I am in the correct way of integrating it.

**Comment by JonnyBurger** [comment]
Answering them on my phone, in case I can unblock you :) 

- Do I need to add it to the player example page ?
> It would be nice to have 1 example of it!

- Do Thumbnail have to handle a composition or a component like the player ?
> Just a component like the Player!

- How could I share logic between Player and Thumbnail? Could I create composable between them?
> I imagine with sharing math functions and contexts

- Is their is way to force the frame context of a composition to be a dedicate frame index?
> Yes, see the implementation of the freeze component! Or maybe, you can use the Freeze component itself!

**Comment by Slashgear** [comment]
@JonnyBurger I will work on that this weekend

File: packages/player-example/src/index.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import {createRoot} from 'react-dom/client';
 import App from './App';
 import CarSlideshow from './CarSlideshow';
 import {VideoautoplayDemo} from './VideoAutoplay';
+import {Thumbnail} from "@remotion/player/dist/Thumbnail";
```
Comment (JonnyBurger) [inline packages/player-example/src/index.tsx]: Should import directly from @remotion/player

File: packages/player/src/Thumbnail.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import type {TComposition, TimelineContextValue} from "remotion";
+import {random, Internals} from "remotion";
+
+export const Thumbnail:React.FC<{
+    composition: TComposition<unknown>;
+    targetHeight: number;
+    targetWidth: number;
+    frameToDisplay: number;
+    style?: CSSProperties
+}> = ({
```
Comment (JonnyBurger) [inline packages/player/src/Thumbnail.tsx]: Ideally the API is consistent with the one of the `<Player>` (props-wise).
The redundancy that is in this component is totally fine though, no need to create crazy abstractions 👍

**Comment by Slashgear** [comment]
@JonnyBurger I try to render the composition from the component inside the thumbnail but I can't figure out why nothing renders. Maybe you have a lead on that ?

**Comment by JonnyBurger** [comment]
@Slashgear The contexts must be setup very carefully. The `<Player>` component might help as a reference since it renders all the React context providers.
There was an error in the browser console that hinted me towards the contexts missing. I've pushed a commit that makes the component render now, although way too big, so `targetWidth` does not yet seem to do anything 😄

**Review by JonnyBurger — APPROVED** [review]
This is awesome 🥳 Very useful, and in fact I only just recently got asked about this feature! Glad I can tell them it's now there!

Appreciate the contribution so much 🙌 💙

How long do you think you spent on this issue? (and the other one?)

**Comment by Slashgear** [comment]
This one, at least 3 hours @JonnyBurger

**Pull request merged**